### DISCUSSION TITLE
How to implement a Tied Autoencoders in Flax?

### DISCUSSION THREAD

**Discussion by avital** [google/flax#2091] [Q&A] [answered]

**Comment by marcvanzee** [comment, marked as answer]
(Tied Autoencoders are autoencoders where the decoder weight is always the transpose of the encoder weight.)

How does Flax's functional approach compare and contrast with a monkey-patching approach as can be seen in some stateful ML libraries?

For reference, see a version of Tied Autoencoders implemented in function core: https://github.com/google/flax/blob/master/tests/core/design/core_tied_autoencoder_test.py
